src/Autosuggest.js

```diff
@@ -140,6 +144,33 @@ export default class Autosuggest extends Component {
     }
 
     this.setState(newState);
+
+    this.onSuggestionUnfocused();
+    this.onSuggestionFocused(sectionIndex, suggestionIndex);
+  }
+
+  onSuggestionFocused(sectionIndex, suggestionIndex) {
+    let suggestion = this.getSuggestion(sectionIndex, suggestionIndex);
+    if(suggestionIndex !== null) {
+      this.props.onSuggestionFocused(suggestion);
+    }
+    lastFocusedSuggestion = suggestion;
+  }
+
+  onSuggestionUnfocused() {
+    if (lastFocusedSuggestion != null) {
+      this.props.onSuggestionUnfocused(lastFocusedSuggestion);
+    }
+    lastFocusedSuggestion = null;
+  }
+
+  onSuggestionSelected(sectionIndex, suggestionIndex, event) {
+    this.onSuggestionUnfocused();
+    this.props.onSuggestionSelected(
+      this.getSuggestion(sectionIndex, suggestionIndex),
+      event
+    );
+    lastFocusedSuggestion = null;
```

Comment:
Why this is required? Wouldn't `lastFocusedSuggestion` be set to `null` when `this.onSuggestionUnfocused()` is called?